Universal FE Randomizer/src/fedata/gba/GBAFEClassData.java

```diff
@@ -17,100 +23,436 @@ public int compare(GBAFEClassData arg0, GBAFEClassData arg1) {
+		int baseSKL = data[13] & 0xFF;
+		if ((baseSKL & 0x80) != 0) {
+			baseSKL |= 0xFFFFFF00;
+		}
+		
+		return baseSKL;
+	}
+
+	public int getBaseSPD() {
+		int baseSPD = data[14] & 0xFF;
+		if ((baseSPD & 0x80) != 0) {
+			baseSPD |= 0xFFFFFF00;
+		}
+		
+		return baseSPD;
+	}
+
+	public int getBaseDEF() {
+		int baseDEF = data[15] & 0xFF;
+		if ((baseDEF & 0x80) != 0) {
+			baseDEF |= 0xFFFFFF00;
+		}
+		
+		return baseDEF;
+	}
+
+	public int getBaseRES() {
+		int baseRES = data[16] & 0xFF;
+		if ((baseRES & 0x80) != 0) {
+			baseRES |= 0xFFFFFF00;
+		}
+		
+		return baseRES;
+	}
+
+	public int getBaseLCK() {
+		return 0;
+	}
 	
-	public int getBaseRankValue();
 	
-	// Miscellaneous
+	public GBAFEStatDto getBases() {
+		return new GBAFEStatDto(getBaseHP(), getBaseSTR(), getBaseSKL(), getBaseSPD(), getBaseDEF(), getBaseRES(), getBaseLCK());
+	}
+
+	public int getMaxHP() {
+		int maxHP = data[19] & 0xFF;
+		if ((maxHP & 0x80) != 0) {
+			maxHP |= 0xFFFFFF00;
+		}
+		
+		return maxHP;
+	}
+
+	public int getMaxSTR() {
+		int maxSTR = data[20] & 0xFF;
+		if ((maxSTR & 0x80) != 0) {
+			maxSTR |= 0xFFFFFF00;
+		}
+		
+		return maxSTR;
+	}
+
+	public int getMaxSKL() {
+		int maxSKL = data[21] & 0xFF;
+		if ((maxSKL & 0x80) != 0) {
+			maxSKL |= 0xFFFFFF00;
+		}
+		
+		return maxSKL;
+	}
+
+	public int getMaxSPD() {
+		int maxSPD = data[22] & 0xFF;
+		if ((maxSPD & 0x80) != 0) {
+			maxSPD |= 0xFFFFFF00;
+		}
+		
+		return maxSPD;
+	}
+
+	public int getMaxDEF() {
+		int maxDEF = data[23] & 0xFF;
+		if ((maxDEF & 0x80) != 0) {
+			maxDEF |= 0xFFFFFF00;
+		}
+		
+		return maxDEF;
+	}
+
+	public int getMaxRES() {
+		int maxRES = data[24] & 0xFF;
+		if ((maxRES & 0x80) != 0) {
+			maxRES |= 0xFFFFFF00;
+		}
+		
+		return maxRES;
+	}
+
+
+	public int getMaxLCK() {
+		return 30;
+	}
 	
-	public int getMOV();
-	public void setMOV(int newMOV);
-	public int getCON();
+	public GBAFEStatDto getCaps() {
+		return new GBAFEStatDto(getMaxHP(), getMaxSTR(), getMaxSKL(), getMaxSPD(), getMaxDEF(), getMaxRES(), getMaxLCK());
+	}
 	
-	// Helpers
-	public Boolean canUseWeapon(GBAFEItemData weapon);
-	public void removeLordLocks();
-	public void overrideAddress(long newAddress);
+	public int getPromoHP() {
+		return data[34] & 0xFF;
+	}
+
+	public int getPromoSTR() {
+		return data[35] & 0xFF;
+	}
+
+	public int getPromoSKL() {
+		return data[36] & 0xFF;
+	}
+
+	public int getPromoSPD() {
+		return data[37] & 0xFF;
+	}
+
+	public int getPromoDEF() {
+		return data[38] & 0xFF;
+	}
+
+	public int getPromoRES() {
+		return data[39] & 0xFF;
+	}
+
+	public int getSwordRank() {
+		return data[44] & 0xFF;
+	}
 	
-	public GBAFEClassData createClone();
+	public GBAFEStatDto getPromoBonuses() {
+		return new GBAFEStatDto(getPromoHP(), getPromoSTR(), getPromoSKL(), getPromoSPD(), getPromoDEF(), getPromoRES(), 0);
+	}
+
+	public void setSwordRank(WeaponRank rank) {
+		FE7Data.Item.FE7WeaponRank fe7Rank = FE7Data.Item.FE7WeaponRank.rankFromGeneralRank(rank);
```

Comment:
If the values are actually the same across all games, then we should make a generic version of the weapon ranks.

Reply:
Comparing the 3 enums shows that FE6 has slightly different values, but FE7 and 8 are Identical, this specific method is universal for all of them however

Reply:
Yeah, I think it's just weird seeing GBAFEClassData hold a reference specifically to FE7 stuff when it theoretically should be completely generic.

Maybe, as you mentioned, the better change would be to make the generic `WeaponRank` enum hold all of the values so you can pull a `.fe6Value`, `.fe7Value`, and `.fe8Value` from it.

Reply:
I do agree that it is weird, I had started looking into trying to make one generic one, but seeing as how there are currently hundreds of references to the 3 classes + the already partially Generic WeaponRank enum, I was kind of hesitant to go through that effort in the scope of this. 

The easiest way of avoiding the FE7 specific reference would I guess be to just mark these methods abstract, considering the class is abstract anyway.

Reply:
Yeah, certainly, it's probably fine for the time being. We can leave it for a larger refactoring effort.